meeting-minutes/2024-11-12.md

```diff
@@ -0,0 +1,392 @@
+
+Kris: I don't think the wire format makes a distinction.
+
+Kris: A question for Spritely: do local-repr vs. remote-repr need a wire format distinction?
+
+Christine: I think the OCapN spec talks about "remote references", but has operations desc:export vs. desc:import-promises vs. desc:import-objects.
+
+Mark: And it's the object identified by the reference that receives and processes messages.
+
+Kris: In the case of a third party...
+
+Mark: References must be recognizably eq to each other, but promises are much weaker. Only fulfillment allows such comparison.
+
+Kris: I propose fast-forwarding to consensus for renaming what I have called "remotable" to "reference".
+
+Mark: A promise is also a reference.
+
+Kris: That implies that one criterion for the name is not differentiating local vs. remote, but rather promise vs. not.
+
+Mark: I like "remotable" with "imported" vs. "exported" qualifiers, particularly because "remotable" makes a good pass-invariant classification (and likewise for "promise").
+
+Kris: Am I correct that Spritely folks don't like the term?
+
+Christine: "Remotable" does not suggest to me exclusion of remote promises. What about "remote actor"?
+
+Mark: From the perspective of CapTP, promises are also actors.
+
+Kris: So we have a perfectly suitable name for promises, but not for referencable objects with convergent pass-invariant identity.
+
+Mark: What I'm stuck on is that the name should identify a pass-invariant category. I like focusing in on the concept of identity to inspire a name.
+
+Jessica: As I understand things, "remotable" encompasses both local objects that can be shared and references to remote objects. So I find the term confusing, and also disagree with "remote actor" for the same reason. It also suffers from encompassing promises, as previously noted.
+
+Baldur: [agrees]
+
+Kris: I propose that the wire protocol and abstract model use "identifier". I know this means something very different in the concrete language implementations, but it does capture the "identity" concept.
+
+Mark: I object.
+
+Christine: In a JavaScript environment, what term would you use for the things can receive messages but aren't promises?
+
+Mark: E used "methodical object", which I would not suggest.
+
+Christine: Kris suggested "target" in chat, which I like.
+
+Kris: I think we're looking at promise vs. target or promise vs. non-promise.
+
+Mark: Is "target" intuitive as the pass-invariant notion, or just for the exported subset?
+
+Kris: My intention was to convey the notion that it is unique, as in the center point of concentric circles 🎯.
+
+Kris: Are there any objections to "imported target" vs. "exported target"?
+
+Richard: To answer Mark, I think "target" does capture the right intuitions.
+
+Kris: I see broad approval now, but we don't need to agree now. I'll update the PR for discussion next month.
+
+### Adjournment
+
+[Jessica and David both volunteer as potential next scribe]
+
+## Chat transcript
```

Comment:
More of a question maybe for the group, rather than direct feedback of something I want changing.

I usually leave the big blue button chat out rather than leave it in the published minutes.